nvmeadm/tests/discovery_test.rs

```diff
@@ -31,3 +171,62 @@ fn connect_test() {
 fn disconnect_test() {
     let _ = disconnect("mynqn");
 }
+
+#[allow(dead_code)]
+fn test_against_real_target() {
+    // Start an SPDK-based nvmf target
+    let _target = NvmfTarget::new(CONFIG_FILE, &TARGET_PORT.to_string());
+
+    // Check that we CANNOT connect to an NQN that does not exist
+    let mut explorer = DiscoveryBuilder::default()
+        .transport("tcp".to_string())
+        .traddr("127.0.0.01".to_string())
+        .trsvcid(TARGET_PORT)
+        .build()
+        .unwrap();
+
+    let pages = explorer.discover();
+    pages.iter().map(|e| println!("{:#?}", e)).for_each(drop);
+
+    let not_expected_to_connect =
+        explorer.connect("an nqn that does not exist");
+    let _ = match not_expected_to_connect {
```

Comment:
since you're not actually user the errors on these expected and not_expected, I'd probably just use is_err and is_ok but that's just my preference :)

Reply:
I left these alone